src/cargo/ops/tree/mod.rs

```diff
@@ -150,6 +150,8 @@ pub fn build_and_print(ws: &Workspace<'_>, opts: &TreeOptions) -> CargoResult<()
     } else {
         ForceAllTargets::No
     };
+    // FIXME: we're creating this here, but build_ctx() is also creating one.
+    // Can we refactor things so that they are shared?
```

Comment:
```suggestion
    // FIXME: we're creating this here, but create_bcx() is also creating one.
    // Can we refactor create_bcx() into multiple functions so that they can be shared?
```